epochStart/metachain/legacySystemSCs.go

```diff
@@ -0,0 +1,1319 @@
+
+	vmInput := &vmcommon.ContractCallInput{
+		VMInput: vmcommon.VMInput{
+			CallerAddr:  s.endOfEpochCallerAddress,
+			Arguments:   [][]byte{s.esdtOwnerAddressBytes, baseIssuingCost, minTokenNameLength, maxTokenNameLength},
+			CallValue:   big.NewInt(0),
+			GasProvided: math.MaxUint64,
+		},
+		Function:      "configChange",
+		RecipientAddr: vm.ESDTSCAddress,
+	}
+
+	output, err := s.systemVM.RunSmartContractCall(vmInput)
+	if err != nil {
+		return err
+	}
+	if output.ReturnCode != vmcommon.Ok {
+		return fmt.Errorf("%w changeESDTOwner should have returned Ok", epochStart.ErrInvalidSystemSCReturn)
+	}
+
+	return s.processSCOutputAccounts(output)
+}
+
+func (s *legacySystemSCProcessor) legacyEpochConfirmed(epoch uint32) {
+	s.flagSwitchJailedWaiting.SetValue(epoch >= s.switchEnableEpoch)
+	log.Debug("systemSCProcessor: switch jail with waiting", "enabled", s.flagSwitchJailedWaiting.IsSet())
+
+	// only toggle on exact epoch. In future epochs the config should have already been synchronized from peers
+	s.flagHystNodesEnabled.SetValue(epoch == s.hystNodesEnableEpoch)
+
+	s.flagChangeMaxNodesEnabled.SetValue(false)
+	for _, maxNodesConfig := range s.maxNodesEnableConfig {
+		if epoch == maxNodesConfig.EpochEnable {
+			s.flagChangeMaxNodesEnabled.SetValue(true)
+			s.maxNodes = maxNodesConfig.MaxNumNodes
+			break
+		}
+	}
+
+	log.Debug("systemSCProcessor: consider also (minimum) hysteresis nodes for minimum number of nodes",
+		"enabled", epoch >= s.hystNodesEnableEpoch)
+
+	// only toggle on exact epoch as init should be called only once
+	s.flagDelegationEnabled.SetValue(epoch == s.delegationEnableEpoch)
+	log.Debug("systemSCProcessor: delegation", "enabled", epoch >= s.delegationEnableEpoch)
+
+	s.flagSetOwnerEnabled.SetValue(epoch == s.stakingV2EnableEpoch)
+	s.flagStakingV2Enabled.SetValue(epoch >= s.stakingV2EnableEpoch && epoch < s.stakingV4InitEnableEpoch)
+	log.Debug("systemSCProcessor: stakingV2", "enabled", epoch >= s.stakingV2EnableEpoch)
+	log.Debug("systemSCProcessor: change of maximum number of nodes and/or shuffling percentage",
+		"enabled", s.flagChangeMaxNodesEnabled.IsSet(),
+		"epoch", epoch,
+		"maxNodes", s.maxNodes,
+	)
+
+	s.flagCorrectLastUnjailedEnabled.SetValue(epoch == s.correctLastUnJailEpoch)
+	log.Debug("systemSCProcessor: correct last unjailed", "enabled", s.flagCorrectLastUnjailedEnabled.IsSet())
+
+	s.flagCorrectNumNodesToStake.SetValue(epoch >= s.correctLastUnJailEpoch)
+	log.Debug("systemSCProcessor: correct last unjailed", "enabled", s.flagCorrectNumNodesToStake.IsSet())
+
+	s.flagESDTEnabled.SetValue(epoch == s.esdtEnableEpoch)
+	log.Debug("systemSCProcessor: ESDT initialization", "enabled", s.flagESDTEnabled.IsSet())
+
+	s.flagSaveJailedAlwaysEnabled.SetValue(epoch >= s.saveJailedAlwaysEnableEpoch)
+	log.Debug("systemSCProcessor: save jailed always", "enabled", s.flagSaveJailedAlwaysEnabled.IsSet())
+
+	s.flagInitStakingV4Enabled.SetValue(epoch == s.stakingV4InitEnableEpoch)
```

Comment:
Setting `flagInitStakingV4Enabled` is done in `legacy`; but it is being used in `systemSCs`. Perhaps move it?

Reply:
no, we need it as well in the legacy.